src/core/weightedGraph.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import {Graph, type GraphJSON} from "./graph";
+import {type Weights as WeightsT, type WeightsJSON} from "./weights";
+import * as Weights from "./weights";
+import {toCompat, fromCompat, type Compatible} from "../util/compat";
+
+export type WeightedGraph = {|
+  +graph: Graph,
+  +weights: WeightsT,
+|};
+
+const COMPAT_INFO = {type: "sourcecred/weightedGraph", version: "0.1.0"};
+
+export type WeightedGraphJSON = Compatible<{|
+  +graphJSON: GraphJSON,
+  +weightsJSON: WeightsJSON,
+|}>;
```

Comment:
I notice we're nesting `Compatible<T>` types here. How does this affect versioning and/or upgrade path of these types?

For example, currently we have:
- Graph v0.8.0
- Weights v0.2.0

Should we create a Graph 0.9.0, which has a sensible upgrade function from v0.8.0.
Would we be able to express that `WeightedGraphJSON` v0.1.0 is the composition of:
- Graph v0.8.0 | v0.9.0
- Weights v0.2.0

Reply:
Interesting question. I suppose that we'd need to bump WGJ to v0.2.0 concurrently with bumping Graph to v0.9.0. Outside the scope of this PR (and not a current priority), but it'd be interesting to see some facilities for handling these nested versioning cases.

Reply:
If `Graph.fromJSON` supports converting v0.8.0 to v0.9.0 transparently
in its `fromJSON` (perhaps via `compat` resolvers), then `WeightedGraph`
won’t need to bump. Otherwise, the `Graph` change is “breaking”, and
`WeightedGraph` will need to bump. As far as I can tell, it’s the same
as normal semver. (No action required.)